firmware/Core/Src/transforms/arrays.c

```diff
@@ -63,6 +63,51 @@ void GEN_uint64_to_str(uint64_t value, char *buffer) {
     buffer[index] = '\0';
 }
 
+/// @brief Converts a int64_t to a string.
+/// @param value The input value to convert.
+/// @param buffer The output buffer to write the string to. MUST BE 32 BYTES LONG.
+void GEN_int64_to_str(int64_t value, char *buffer) {
+    // Ensure the buffer is valid and large enough
+    if (buffer == NULL) {
+        return;
+    }
+
+    char temp[32];  // Temporary array to hold digits
+    int index = 0;
+    int is_negative = 0;  // Flag to track if the number is negative
```

Comment:
What about `const uint8_t is_negative = (value < 0);`

Reply:
This makes sense.